augur/measurements/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""
+Create JSON files suitable for visualization within the measurements panel of Auspice.
+"""
+import argparse
+import importlib
+from augur.utils import first_line
+
+SUBCOMMAND_MODULE_NAMES = [
+    "export",
+    "concat",
+]
+
+SUBCOMMANDS = [(c.replace("_", "-"), importlib.import_module(f'augur.measurements.{c}')) for c in SUBCOMMAND_MODULE_NAMES]
+
+
+def register_arguments(parser):
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subparsers.required = True
+
+    for subcommand_name, subcommand in SUBCOMMANDS:
+        # Add a subparser for each subcommand
+        subparser = subparsers.add_parser(
+            subcommand_name,
+            help        = first_line(subcommand.__doc__),
+            description = subcommand.__doc__
+        )
+
+        # Allows us to run subcommands directly with `args.__subcommand__.run()`
+        subparser.set_defaults(__subcommand__ = subcommand)
+
+        # Let the subcommand register arguments on its subparser
+        subcommand.register_arguments(subparser)
+
+        # Use the same formatting class for every command for consistency.
+        # Set here to avoid repeating it in every command's register_parser().
+        subparser.formatter_class = argparse.ArgumentDefaultsHelpFormatter
```

Comment:
This is duplicated from [`augur/__init__.py`](https://github.com/nextstrain/augur/blob/1b2d1e2e6f32c7d54f72011b1879e582e0667b80/augur/__init__.py#L59-L73) and I can imagine wanting to use the same block for other commands that implement subcommands (e.g. `augur curate`).

Here's an idea to make it re-usable: https://github.com/victorlin/augur/commit/17194d1cc48dff0e96b5e2b224e05de2d286fc6a

Reply:
Oh, I didn't see https://github.com/nextstrain/augur/pull/1002#discussion_r917147387 and cb7970b9a16f0c72eae324885d9a6315e4970eb0 until now... I still think having a shared `add_subparsers()` is reasonable, unless you think other commands might implement this differently?

Reply:
Ah, thanks both for confirming the initial idea was in the right direction. 

Looking at how this is handled in Nextstrain CLI, I _do_ think it's a bit weird in augur that we only allow commands to register arguments instead of their own subparser. I think this was what made me think `add_subparsers()` was a bit convoluted and drop cb7970b9a16f0c72eae324885d9a6315e4970eb0.

Reply:
Hmm okay, I'll do some more refactoring...By having each command register it's own subparser, we also would remove the need to parse or declare a `command_name` property.

Reply:
Addressed in 86707ef